presentation/json.go

```diff
@@ -0,0 +1,140 @@
+package presentation
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/box/memsniff/analysis"
+	"os"
+	"time"
+)
+
+func (u *uiContext) runLoggingEventLoop() error {
+	updateTick := time.NewTicker(u.interval)
+	defer updateTick.Stop()
+	if err := u.update(); err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-updateTick.C:
+			if err := u.updateReport(); err != nil {
+				return err
+			}
+
+		case msg := <-u.msgChan:
+			u.logger.Log(msg)
+		}
+	}
+}
+
+func (u *uiContext) updateReport() error {
+	rep := u.analysis.Report(!u.cumulative)
+	rep.SortBy(-2)
+
+	numKeysSeen := len(rep.Rows)
+	totalBandwidthUsed := totalBytesUseForKeys(rep.Rows)
+
+	s := u.statProvider()
+
+	u.truncateResultsToMaxAndTopX(&rep)
+	u.prevReport = rep
+	numReportedKeys := len(rep.Rows)
+	reportedKeysBandwidth := totalBytesUseForKeys(rep.Rows)
+
+	f, err := u.openFile(u.outputFile)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	reportJson, _ := formatReportAsJson(u.prevReport, s, numKeysSeen, totalBandwidthUsed,numReportedKeys,reportedKeysBandwidth)
+
+	u.Log(fmt.Sprintf("All_Keys(Bandwidth=%d,Num=%d) Reported_Keys(Bandwidth=%d (%f%%),Num=%d) Incremental(Packets: %d Responses: %d %s) Cumulative(Packets: %d Responses: %d %s)",
```

Comment:
This isn't a useful format either for humans (why are there underscores?) or machines. Pick one or the other.

Reply:
This format is purely for humans.  I find is readable but I guess it is subjective.
e.g. 
```
2020/01/09 13:45:56.096793 All_Keys(Bandwidth=9291776,Num=20) Reported_Keys(Bandwidth=7793664 (83.877011%),Num=1) Incremental(Packets: 31000 Responses: 7744 Dropped: 0+0+0=0 ( 0.00%)) Cumulative(Packets: 359000 Responses: 89640 Dropped: 0+0+0=0 ( 0.00%))
2020/01/09 13:45:57.095868 All_Keys(Bandwidth=9994240,Num=20) Reported_Keys(Bandwidth=8445952 (84.508197%),Num=2) Incremental(Packets: 33000 Responses: 8240 Dropped: 0+0+0=0 ( 0.00%)) Cumulative(Packets: 392000 Responses: 97880 Dropped: 0+0+0=0 ( 0.00%))
```

Is the objection just to the underscores?